src/ordered_set_lang.ml

```diff
@@ -45,19 +45,30 @@ module Parse = struct
           | _ ->
             elt >>| fun x -> Element x
         end
-      | List (_, Atom (_, A ":include") :: _) -> inc
-      | List _ -> enter (many [])
-    and many acc =
+      | List (_, Atom (loc, A s) :: _) -> begin
+          match s, kind with
+          | ":include", _ -> inc
+          | s, Dune when s <> "" && s.[0] <> '-' && s.[0] <> ':' ->
+            Loc.fail loc
+              "This atom must be quoted because it is the first element \
+               of a list and doesn't start with - or :"
```

Comment:
Can you document this syntax in the "Ordered set language" section in doc/jbuild.rst?

Reply:
done